### PR TITLE
Implement default description for Community/Collection for OpenSearch

### LINKED ISSUE #8728: OpenSearch endpoint can't generate RSS 2.0 feeds for Community/Collection without description
**Describe the bug**
When attempting to generate an RSS feed for a Community/Collection without `dc.description.abstract` metadata, the `/opensearch` endpoint fails with the following error
```
Invalid rss_2.0 feed, missing channel description
```

**To Reproduce**
1. Find (or create) a Community/Collection without `dc.description.abstract` metadata on https://demo7.dspace.org
2. Request `https://api7.dspace.org/server/opensearch/search?format=rss&scope=<UUID>&query=*`
   * The request will fail with HTTP 500, you'll see the above error message in the logs
  

**Expected behavior**
The feed should still work.

**Proposed solution**
Fall back to a default string (e.g. 'No description') in order to conform to the RSS 2.0 spec.

**Workaround**
Just add a `dc.description.abstract` to the Community/Collection.

**Related work**
Link to any related tickets or PRs here.

### PR DESCRIPTION
**Pull request by CVamsi27**
## References
* Fixes #8728

## Description
Implemented default description as "No Description" for Community/Collection.

## Instructions for Reviewers
- Implemented default description as "No Description" for Community/Collection.
- Now if we ping `https://api7.dspace.org/server/opensearch/search?format=rss&scope=<UUID>&query=*` with a Community/Collection - UUID we will get a 200 response with default description.

List of changes in this PR:
* Implemented default description as "No Description" for Community/Collection.
* Added test case to to check default description.

## Checklist
_This checklist provides a reminder of what we are going to look for when reviewing your PR. You need not complete this checklist prior to creating your PR (draft PRs are always welcome). If you are unsure about an item in the checklist, don't hesitate to ask. We're here to help!_

- [x] My PR is small in size (e.g. less than 1,000 lines of code, not including comments & integration tests). Exceptions may be made if previously agreed upon.
- [x] My PR passes Checkstyle validation based on the [Code Style Guide](https://wiki.lyrasis.org/display/DSPACE/Code+Style+Guide).
- [x] My PR includes Javadoc for _all new (or modified) public methods and classes_. It also includes Javadoc for large or complex private methods.
- [x] My PR passes all tests and includes new/updated Unit or Integration Tests based on the [Code Testing Guide](https://wiki.lyrasis.org/display/DSPACE/Code+Testing+Guide).
- [x] If my PR includes new libraries/dependencies (in any `pom.xml`), I've made sure their licenses align with the [DSpace BSD License](https://github.com/DSpace/DSpace/blob/main/LICENSE) based on the [Licensing of Contributions](https://wiki.lyrasis.org/display/DSPACE/Code+Contribution+Guidelines#CodeContributionGuidelines-LicensingofContributions) documentation.
- [x] If my PR modifies REST API endpoints, I've opened a separate [REST Contract](https://github.com/DSpace/RestContract/blob/main/README.md) PR related to this change.
- [x] If my PR includes new configurations, I've provided basic technical documentation in the PR itself.
- [x] If my PR fixes an issue ticket, I've [linked them together](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue).

### CODE REVIEW

**Comment by CVamsi27** [comment]
Please review this pull request and let me know if any change is required.

**Comment by CVamsi27** [comment]
Can I also add fixes to this [issue](https://github.com/DSpace/DSpace/issues/8671) in the same PR?

**Comment by tdonohue** [comment]
@CVamsi27 : It's better to create separate PRs if the issues you are fixing are unrelated (and #8671 doesn't seem related to OpenSearch).  That said, if two issues are related to the same feature/component, it's fine to fix both in a single PR.

**Review by tdonohue — CHANGES_REQUESTED** [review]
@CVamsi27 : Reviewed & tested this today. Overall it works. However, I think we should make sure the "No Description" text can be translated by moving it into `Messages.properties`.  Once this is done, I'd be +1 this PR.

Thanks!  See inline comments below for how to move this into our messages file.

File: dspace-api/src/main/java/org/dspace/app/util/SyndicationFeed.java [hunk 1]
```diff
@@ -230,6 +231,12 @@ public void populate(HttpServletRequest request, Context context, IndexableObjec
         }
         feed.setTitle(labels.containsKey(MSG_FEED_TITLE) ?
                           localize(labels, MSG_FEED_TITLE) : defaultTitle);
+
+        if (defaultDescriptionField == null || defaultDescriptionField == "") {
+            defaultDescriptionField = "No Description";
```
[the remark this answers is not in the document]
Comment (CVamsi27) [reply dspace-api/src/main/java/org/dspace/app/util/SyndicationFeed.java]: Updated

**Review by tdonohue — APPROVED** [review]
👍 Thanks @CVamsi27 !  Retested briefly today and it's working well. Code looks good now too.

**Pull request merged**